pymess/models/push.py

```diff
@@ -0,0 +1,77 @@
+from jsonfield.fields import JSONField
+
+from chamber.utils.datastructures import ChoicesNumEnum
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from pymess.config import get_push_notification_sender, settings
+
+from .common import BaseAbstractTemplate, BaseMessage, BaseRelatedObject
+
+__all__ = (
+    'AbstractPushNotificationMessage',
+    'AbstractPushNotificationTemplate',
+    'PushNotificationMessage',
+    'PushNotificationMessageRelatedObject',
+    'PushNotificationTemplate',
+)
+
+
+class AbstractPushNotificationMessage(BaseMessage):
+
+    STATE = ChoicesNumEnum(
+        ('WAITING', _('waiting'), 1),
+        ('SENT', _('sent'), 2),
+        ('ERROR', _('error'), 3),
+        ('DEBUG', _('debug'), 4),
+    )
+
+    template = models.ForeignKey(settings.PUSH_NOTIFICATION_TEMPLATE_MODEL, verbose_name=_('template'), blank=True,
+                                 null=True, on_delete=models.SET_NULL, related_name='push_notifications')
+    state = models.PositiveIntegerField(verbose_name=_('state'), null=False, blank=False, choices=STATE.choices,
+                                        editable=False)
+    heading = models.TextField(verbose_name=_('heading'))
```

Comment:
heading v sobě obsahuje i text?

Reply:
Push notifikace mají nějaký heading a content. To je ono.

Reply:
jo content je v předkovi už chápu